### PR TITLE
feat: Add callback `crypto/v1/is_certificate_trusted`

### PR DESCRIPTION
**Pull request by viccuad**
## Description

Fix https://github.com/kubewarden/policy-evaluator/issues/200
Needs https://github.com/kubewarden/policy-sdk-rust/pull/69
Needs https://github.com/kubewarden/policy-sdk-rust/pull/71

Given a cert and a cert chain, verifies the validity of the cert against
the passed cert chain.
If no chain is passed, only the cert validity against the current time
is checked.

Add new module `callback_handler::crypto`, that implements the cert
verification.

## Test

Added new unit test for the callback request and response.

### CODE REVIEW

**Review by jvanz — APPROVED** [review]

File: src/callback_handler/crypto.rs [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+
+/// verify_certificate verifies the validity of the certificate, and if it is
+/// trusted with the provided certificate chain.
+/// If the provided certificate chain is empty, it is treated as trusted.
+pub fn verify_certificate(req: CertificateVerificationRequest) -> Result<bool> {
+    // verify validity:
+    let pc = match req.cert.encoding {
+        CertificateEncoding::Pem => {
+            let pem_str = String::from_utf8(req.cert.data.clone())
+                .map_err(|_| anyhow!("Certificate is not PEM encoded"))?;
```
Comment (flavio) [inline src/callback_handler/crypto.rs:22]:
```suggestion
                .map_err(|_| anyhow!("Certificate PEM data is not UF8 encoded"))?;
```

File: src/callback_handler/crypto.rs [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+    /// Build a `CertificatePool` instance using the provided list of [`Certificate`]
+    fn from_certificates(certs: &[Certificate]) -> Result<Self> {
+        let mut trusted_roots = vec![];
+        let mut intermediates = vec![];
+
+        for c in certs {
+            let pc = match c.encoding {
+                CertificateEncoding::Pem => {
+                    let pem_str = String::from_utf8(c.data.clone())
+                        .map_err(|_| anyhow!("Certificate is not PEM encoded"))?;
```
Comment (flavio) [inline src/callback_handler/crypto.rs:83]:
```suggestion
                        .map_err(|_| anyhow!("Certificate PEM data is not UTF8 encoded"))?;
```

**Review by flavio — COMMENTED** [review]
I made some changes by doing a commit into this branch, I hope you won't mind. I had to try out the code before suggesting it. Once I got it working it was easier to submit it rather then go back and use `suggestion` boxes.

I've also left some comments. Overall looks good to me.

**Comment by flavio** [comment]
@viccuad as for the unit tests: I'm fine with the hard coded certs. Are you fine with them too or do you still want to generate them programmatically?

File: Cargo.toml [hunk 3]
```diff
@@ -38,8 +39,11 @@ validator = { version = "0.16", features = ["derive"] }
...
 wapc = "1.0.0"
 wasmtime-provider = { version = "1.3.2", features = ["cache"] }
+picky = { version = "7.0.0-rc.3", default-features = false, features = [ "x509", "ec", "chrono_conversion" ] }
+chrono = "0.4.23"
 
 
 [dev-dependencies]
 assert-json-diff = "2.0.2"
 k8s-openapi = { version = "0.16.0", default-features = false, features = ["v1_24"] }
+chrono = "0.4.20"
```
[the remark this answers is not in the document]
Comment (viccuad) [reply Cargo.toml]: Doh, done in https://github.com/kubewarden/policy-evaluator/pull/208/commits/0a9be6a46f49b98a9927a6d3a52ba5ea2ddbb64b

**Comment by viccuad** [comment]
fixup of commit consuming sdk from my fork (sets me as committer of refactor commit, leaves author intact), bumped version.

**Pull request merged**